azclishell/app.py

```diff
@@ -393,7 +393,7 @@ def _special_cases(self, text, cmd, outside):
         if self.default_command:
             cmd = self.default_command + " " + cmd
 
-        if text.strip() == "quit" or text.strip() == "exit":
+        if cmd.strip() == "quit" or cmd.strip() == "exit":
```

Comment:
A little. The text variable should always be what's represented on the line, but the cmd is reformatted for what the shell needs for information